lib/game.rb

```diff
@@ -41,46 +52,84 @@ def setup
     submarine_coordinates = gets.chomp.split(',')
 
     until_valid_placement(@player.submarine, submarine_coordinates)
-
-    puts @player.board.render(true)
   end
   
   def turn
+    both_boards_rendered
+    puts "Your turn:"
     puts "Enter coord to fire on"
     coord = gets.chomp.upcase
     @player.fire_upon_computer(coord, @computer.board)
-    # The above was just so I could test the #fire_upon_computer in Player class. You can remove/update it if you want
-    
-    # At some point in here, we will call fire_upon_player(@player.board) to fire upon the player's individual cell within their board instance
-    # fire_upon_player(@player.board) should return the result of the cell being fired upon (was it a hit or miss?)
-    
-    # At some point in here, we will call fire_upon_computer(coordinate, @computer.board) the coordinate argument = the coordinate that the player chose to fire upon
-
+    # right now, both the computer and the player can win (if it comes down to the very last turn)
+    # perhaps we place a check in here to end the turn before player or computer can choose in the event they have been sunk
+    puts "My turn:"
+    @computer.fire_upon_player(@player.board)
+    # Also: it might be better if we use interpolation to say whether the shot was a "hit" or "miss".
+    # for some reason the final shot returns "miss" instead of a hit, even though that shot ends the game.
+    # MAYBE THIS IS HAPPENING BECAUSE THE FINAL SHOT DOES NOT REGISTER AS A HIT, BECAUSE IT IS CHANGING THE STATUS TO SUNK?
   end
 
+  # results method gives different message based on which board has all sunken ships
   def results
-    # if computer.ships.all?(sunk?)
-    # return "You win!"
-
-    # if player.ships.all?(sunk?)
-    # return "I won""
+    # it might be nice to show the board at the end of the game, to look at
+    puts "=🌊=🌊=🌊=🌊=🌊=🌊=🌊=🌊=🌊"
+    puts "The results:"
+    both_boards_rendered
+
+    if computer_ships_all_sunk?
+      puts "You win!"
+      "You win!"
```

Comment:
Do we need to return this string again if it's already being `put` to the terminal on the line above?

Reply:
No, we do not. I was using `puts "Hit!"` or `puts "Miss!"` during the @ship.hit method so I could see it happen during the turn. I think we should choose which we'd rather use for the end game result. (Interpolation, or simply using put)